pkg/agent/controller/noderoute/node_route_controller_test.go

```diff
@@ -228,3 +231,153 @@ func TestIPInPodSubnets(t *testing.T) {
+			Name: "xyz-k8s-0-1",
+		},
+		Spec: corev1.NodeSpec{
+			PodCIDR:  podCIDR.String(),
+			PodCIDRs: []string{podCIDR.String()},
+		},
+		Status: corev1.NodeStatus{
+			Addresses: []corev1.NodeAddress{
+				{
+					Type:    corev1.NodeInternalIP,
+					Address: nodeIP1.String(),
+				},
+			},
+		},
+	}
+
+	c.clientset.CoreV1().Nodes().Create(context.TODO(), node1, metav1.CreateOptions{})
+	c.ovsClient.EXPECT().DeletePort("123").Times(1)
+
+	err := c.removeStaleTunnelPorts()
+	assert.NoError(t, err)
+}
+
+func TestCreateIPSecTunnelPort(t *testing.T) {
+	c, closeFn := setup(t, []*interfacestore.InterfaceConfig{
+		{
+			Type:          interfacestore.TunnelInterface,
+			InterfaceName: "mismatchedname",
+			TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+				NodeName: "xyz-k8s-0-2",
+				Type:     "vxlan",
+				PSK:      "changeme",
+				RemoteIP: nodeIP2,
+			},
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				PortUUID: "123",
+			},
+		},
+		{
+			Type:          interfacestore.TunnelInterface,
+			InterfaceName: util.GenerateNodeTunnelInterfaceName("xyz-k8s-0-3"),
+			TunnelInterfaceConfig: &interfacestore.TunnelInterfaceConfig{
+				NodeName: "xyz-k8s-0-3",
+				Type:     "vxlan",
+				PSK:      "changeme",
+				RemoteIP: net.ParseIP("10.10.10.1"),
+			},
+			OVSPortConfig: &interfacestore.OVSPortConfig{
+				PortUUID: "abc",
+				OFPort:   int32(5),
+			},
+		},
+	})
+
+	defer closeFn()
+	defer c.queue.ShutDown()
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	node1PortName := util.GenerateNodeTunnelInterfaceName("xyz-k8s-0-1")
+	c.ovsClient.EXPECT().CreateTunnelPortExt(
+		node1PortName, ovsconfig.TunnelType("vxlan"), int32(0),
+		false, "", nodeIP1.String(), "changeme",
+		map[string]interface{}{ovsExternalIDNodeName: "xyz-k8s-0-1"}).Times(1)
+	c.ovsClient.EXPECT().GetOFPort(node1PortName).Return(int32(1), nil)
+
+	tests := []struct {
+		name       string
+		nodeName   string
+		peerNodeIP net.IP
+		wantErr    bool
+		want       int32
+	}{
+		{
+			name:       "good path",
+			nodeName:   "xyz-k8s-0-1",
+			peerNodeIP: nodeIP1,
+			wantErr:    false,
+			want:       1,
+		},
+		{
+			name:       "hit cache but interface name changed for the same node",
+			nodeName:   "xyz-k8s-0-2",
+			peerNodeIP: nodeIP2,
+			wantErr:    true,
+		},
+		{
+			name:       "hit cache and return directly",
+			nodeName:   "xyz-k8s-0-3",
+			peerNodeIP: net.ParseIP("10.10.10.1"),
+			wantErr:    false,
+			want:       5,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := c.createIPSecTunnelPort(tt.nodeName, tt.peerNodeIP)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Controller.createIPSecTunnelPort() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("Controller.createIPSecTunnelPort() got %v,want = %v", got, tt.want)
+				return
+			}
```

Comment:
can we use testify assertions for these tests like we do in other Antrea unit tests? `assert` assertions will continue the test on error, while `require` assertions will abort the test on error

Reply:
done